susemanager-utils/susemanager-sls/salt/uptodate.sls

```diff
@@ -49,20 +49,20 @@ mgr_keep_system_up2date_pkgs:
 {%- if salt['pillar.get']('mgr_reboot_if_needed', True) and salt['pillar.get']('custom_info:mgr_reboot_if_needed', 'true')|lower in ('true', '1', 'yes', 't') %}
 mgr_reboot_if_needed:
   cmd.run:
-    - name: shutdown -r +5
+    - name: /usr/sbin/shutdown -r +5
     - require:
       - pkg: mgr_keep_system_up2date_pkgs
 {%- if grains['os_family'] == 'RedHat' and grains['osmajorrelease'] >= 8 %}
-    - onlyif: 'dnf -q needs-restarting -r; [ $? -eq 1 ]'
+    - onlyif: '/usr/bin/dnf -q needs-restarting -r; [ $? -eq 1 ]'
 {%- elif grains['os_family'] == 'RedHat' and grains['osmajorrelease'] >= 7 %}
-    - onlyif: 'needs-restarting -r; [ $? -eq 1 ]'
+    - onlyif: '/usr/bin/needs-restarting -r; [ $? -eq 1 ]'
 {%- elif grains['os_family'] == 'Debian' %}
     - onlyif:
-      - test -e /var/run/reboot-required
+      - /usr/bin/test -e /var/run/reboot-required
 {%- elif grains['os_family'] == 'Suse' and grains['osmajorrelease'] <= 12 %}
     - onlyif:
-      - test -e /boot/do_purge_kernels
+      - /usr/bin/test -e /boot/do_purge_kernels
 {%- else %}
-    - onlyif: 'zypper ps -s; [ $? -eq 102 ] || [ {{ patch_need_reboot }} -eq 0 ]'
+    - onlyif: '/usr/bin/zypper ps -s; [ $? -eq 102 ] || [ {{ patch_need_reboot }} -eq 0 ]'
```

Comment:
```
$> which [
/usr/bin/[
```
:-)
The closing is not a command. Maybe to not create heavy confusion we should use `/usr/bin/test` instead of `[]`